tslib.es6.js

```diff
@@ -350,17 +350,17 @@ export function __disposeResources(env) {
 
 export function __rewriteRelativeImportExtension(path, preserveJsx) {
     if (typeof path === "string" && path[0] === "." && (path[1] === "/" || path[1] === "." && path[2] === "/")) {
-        if (path.substring(path.length - 4) === ".tsx") {
+        if (path.substring(path.length - 4).toLowerCase() === ".tsx") {
             return path.substring(0, path.length - 4) + (preserveJsx ? ".jsx" : ".js");
         }
-        if (path.substring(path.length - 3) === ".ts") {
-            var dot = path.lastIndexOf(".", path.length - 4);
-            if (dot >= 0 && (path.substring(dot - 2, dot) === ".d" || path.substring(dot, dot + 2) === ".d")) {
-                return path;
+        var extMatch = path.match(/\.[cm]?ts$/i);
+        if (extMatch) {
+            var ext = extMatch[0].toLowerCase();
+            var dot = path.lastIndexOf(".", path.length - (ext.length + 1));
+            if (dot < 0 || !(path.substring(dot - 2, dot).toLowerCase() === ".d" || path.substring(dot, dot + 2).toLowerCase() === ".d")) {

```

Comment:
The string matching logic here is fairly confusing. Is there a reason you don't just check `path.substring(extMatch.index - 2, extMatch.index).toLowerCase() === ".d"`?

Reply:
It’s checking both for `.d.[cm]?ts` and e.g. `.d.css.ts`